api/src/main/java/javax/enterprise/inject/spi/container/Container.java

```diff
@@ -0,0 +1,341 @@
+ * </pre>
+ * <p>
+ * <p>
+ * <p>
+ * In the same manner, it is possible to explicitly declare interceptors, decorators, extensions and implementation specific options using the builder.
+ * </p>
+ * <p>
+ * <pre>
+ * Container container = ContainerProvider.getContainer()
+ *    .disableDiscovery()
+ *    .packages(Main.class, Utils.class)
+ *    .interceptors(TransactionalInterceptor.class)
+ *    .property("property", true);
+ * CDI<Object> cdi = container.initialize();
+ * </pre>
+ *
+ * @since 2.0
+ * @author Antoine Sabot-Durand
+ * @author Martin Kouba
+ * @author John D. Ament
+ */
+public interface Container {
+
+    /**
+     * Define the set of bean classes for the synthetic bean archive.
+     *
+     * @param classes classes to add to the synthetic bean archive
+     * @return self
+     */
+    Container beanClasses(Class<?>... classes);
+
+    /**
+     * Add a bean class to the set of bean classes for the synthetic bean archive.
+     *
+     * @param beanClass class to add to the synthetic archive
+     * @return self
+     */
+    Container addBeanClass(Class<?> beanClass);
+
+    /**
+     * All classes from the packages of the specified classes will be added to the set of bean classes for the synthetic bean archive.
+     * <p>
+     * <p>
+     * Note that the scanning possibilities are limited. Therefore, only directories and jar files from the filesystem are supported.
+     * </p>
+     * <p>
+     * <p>
+     * Scanning may also have negative impact on bootstrap performance.
+     * </p>
+     *
+     * @param packageClasses classes whose packages will be added to the synthetic bean archive
+     * @return self
+     */
+    Container packages(Class<?>... packageClasses);
+
+    /**
+     * Packages of the specified classes will be scanned and found classes will be added to the set of bean classes for the synthetic bean archive.
+     *
+     * @param scanRecursively should subpackages be scanned or not
+     * @param packageClasses classes whose packages will be scanned
+     * @return self
+     */
+    Container addPackages(boolean scanRecursively, Class<?>... packageClasses);
+
+    /**
+     * A package of the specified class will be scanned and found classes will be added to the set of bean classes for the synthetic bean archive.
+     *
+     * @param scanRecursively should subpackages be scanned or not
+     * @param packageClass class whose package will be scanned
+     * @return self
+     */
+    Container addPackage(boolean scanRecursively, Class<?> packageClass);
+
+
+    /**
+     * Returns a {@link AnnotatedTypeConfigurator} based on the provided type
+     *
+     * @param type that the {@link AnnotatedType} to configure represents
+     * @param <T> required type
+     * @return an AnnotatedTypeConfigurator whose configuration will be added as new AnnotatedType
+     */
+    <T> AnnotatedTypeConfigurator<T> configureAnnotatedType(Class<T> type);
```

Comment:
I think the idea was to enable setup of AT without need to create an extension. It's questionable. Maybe this could be available via some extension builder as well.